### PR TITLE
docs: update references to sequence counters to consistently start counting from 0

### PR DESCRIPTION
**Pull request by sarahboyce**
### Summary
The sequence counter starts from 0 but there are some instances in the docs that imply it starts from 1.
Updated these instances as suggested by this issue: https://github.com/FactoryBoy/factory_boy/issues/848

Tried to add some clarifying notes, please let me know if I've made things worse and not better, I won't be offended :smile: 
This is also my first contribution, happy to be pointed in a different direction on how to help if something else is more useful :+1:

### CODE REVIEW

File: docs/introduction.rst [hunk 1]
```diff
@@ -99,11 +99,18 @@ This is achieved with the :class:`~factory.Sequence` declaration:
 
 .. code-block:: pycon
 
+    >>> # The sequence counter starts at 0 by default
     >>> UserFactory()
     <User: user0>
     >>> UserFactory()
     <User: user1>
 
+    >>> # The Sequence can be overridden, this still increments the sequence counter
```
Comment (rbarrois) [inline docs/introduction.rst:108]: I would write this as `# A value can be provided for a sequence-driven field, but this still increments the sequence counter`

**Review by rbarrois — APPROVED** [review]
Hey, thanks a lot for the amazing work on this PR!

I especially appreciate the included cross-references, those add lots of value to the doc ;)

File: docs/introduction.rst (hunk 1 above)
Comment (sarahboyce) [reply docs/introduction.rst]: image
Is this ok?

Comment (rbarrois) [reply docs/introduction.rst]: That's awesome, thanks!

**Review by rbarrois — APPROVED** [review]
It's all good!

The final step is to add yourself to the [CREDITS file](https://github.com/FactoryBoy/factory_boy/blob/master/CREDITS) — or I can do it myself if you prefer ;)

**Review by rbarrois — APPROVED** [review]

**Comment by rbarrois** [comment]
Thanks a lot ;)

**Pull request merged**